Nosco: Add idea content and fields

jkxyz opened the pull request:
Update the Nosco connector to expose additional fields.

jkxyz committed:
Add idea content and fields

vmanoharas commented:
Hi @jkxyz, 

Can you please correct the folder structure before proceeding the review?

![image](https://github.com/user-attachments/assets/d34622ba-7f58-4379-a620-0be8bd4c3d24)

jkxyz commented:
Hi @vmanoharas 

I see a number of errors in the CODEOWNERS file that are already present in the dev/master branch, related to users who seem to no longer exist. However, I do see that my change adds an additional error: 

```
 Unknown owner on line 64: make sure @jkxyz exists and has write access to the repository 
```

Can this be remedied by giving me write access to the repo? 

I've also updated my branch to add an extra newline at the end of that file. Other than that, the syntax of that file appears to be correct.

Do you need anything else before proceeding with your review?

jkxyz commented:
Hi @vmanoharas, 

Please let me know if you need any further changes to this PR before reviewing!

vmanoharas commented:
Hello @jkxyz, 

This is your development branch, and you are responsible for managing it. Please correct the issue before we proceed with the review.

jkxyz commented:
@vmanoharas Can you please be more specific about the issue that needs to be fixed? As far as I can tell, the directory structure is correct.

vmanoharas commented:
Hello @jkxyz, 

Please look into the image below, the folder structure is not as expected. Please follow the instructions in the below link and remove the folders which are not required.  

![image](https://github.com/user-attachments/assets/f83d3f38-4afc-4b8e-9599-dfe66f0737cd)

https://github.com/microsoft/PowerPlatformConnectors

Additionally, kindly review the below swagger validation errors and confirm/fix them if the existing flow will not be broken. 

##[error]The output field has been added when additional properties are not allowed. This will break validation for existing flows. ValuePath: operation/GetIdea/responses/200/schema/fields Source: NewSwagger
##[error]The output field has been added when additional properties are not allowed. This will break validation for existing flows. ValuePath: operation/GetIdea/responses/200/schema/content Source: NewSwagger
##[error]The output field has been added when additional properties are not allowed. This will break validation for existing flows. ValuePath: operation/Ideas/responses/200/schema/nodes/items/fields Source: NewSwagger
##[error]The output field has been added when additional properties are not allowed. This will break validation for existing flows. ValuePath: operation/Ideas/responses/200/schema/nodes/items/content Source: NewSwagger

vmanoharas commented:
Additionally, observed one more commit showed in the below screenshot, this wouldn't be required to commit.  Please revert the same to correct the folder structure. 

![image](https://github.com/user-attachments/assets/c29a09a3-45e9-4e12-8eab-bcb004baafa0)

jkxyz commented:
Hi @vmanoharas 

Thanks for taking the time to clarify for me!

> Additionally, kindly review the below swagger validation errors and confirm/fix them if the existing flow will not be broken.
> 
> ##[error]The output field has been added when additional properties are not allowed. This will break validation for existing flows. ValuePath: operation/GetIdea/responses/200/schema/fields Source: NewSwagger ##[error]The output field has been added when additional properties are not allowed. This will break validation for existing flows. ValuePath: operation/GetIdea/responses/200/schema/content Source: NewSwagger ##[error]The output field has been added when additional properties are not allowed. This will break validation for existing flows. ValuePath: operation/Ideas/responses/200/schema/nodes/items/fields Source: NewSwagger ##[error]The output field has been added when additional properties are not allowed. This will break validation for existing flows. ValuePath: operation/Ideas/responses/200/schema/nodes/items/content Source: NewSwagger

I can confirm that these additional fields are expected, and there are no breaking changes in the API or connector.

> Additionally, observed one more commit showed in the below screenshot, this wouldn't be required to commit. Please revert the same to correct the folder structure.
> 
> ![image](https://private-user-images.githubusercontent.com/137378994/394833468-c29a09a3-45e9-4e12-8eab-bcb004baafa0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNjAzNTYsIm5iZiI6MTczNDM2MDA1NiwicGF0aCI6Ii8xMzczNzg5OTQvMzk0ODMzNDY4LWMyOWEwOWEzLTQ1ZTktNGUxMi04ZWFiLWJjYjAwNGJhYWZhMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxNlQxNDQwNTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNDIxMGNhMTc5M2ZlNmViNzUwMzgyYzlmYzJmZmM3NGU4YzY5MDVkZWZmYWI1ZDYzZTk2ODc0MzI0ZjJhNTAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4ikmCqvX7BI059gdsU8YI-9LxLKhD1Qx5zJqgiUdLX0)

I've removed this commit from the branch. I added it as described in the [certification docs](https://learn.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#essential-program-notes), to allow Nosco to manage future changes to our connector ourselves. 

Is it possible to add me to the code owners for our connector in a separate PR?

Please let me know if further changes are required before merging.

vmanoharas commented:
Thank you @jkxyz for addressing the issues. After running the swagger validation recieved same errors. So I am approving this PR. Thank you for working with us.

vmanoharas reviewed (approved):
Dear Partner, 

Congratulations your PR is approved !! We are proud to announce the Brand-New Certification experience for you to certify Power Platform Copilot Connectors & Plugins. Read the blog here- [Announcing Partner Center to certify and publish Power Platform Copilot Connectors and Plugins - Microsoft Power Platform Blog](https://www.microsoft.com/en-us/power-platform/blog/power-automate/announcing-partner-center-to-certify-and-publish-power-platform-copilot-connectors-and-plugins/)

As next steps: 

1.	Learn about the new certification experience- [Get your Power Platform connector and plugin certified - Overview](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fsubmit-certification&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390456966%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=uWDvs%2BmTbozw1oPc%2FNDNWJ%2FHgnUSfFlRCB1T8UHBh3A%3D&reserved=0)
2.	Package your connector and plugin files- [Prepare Power Platform connector and plugin files for certification](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-submission&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390465418%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=pDuv4bnya1eP3Rrs3p4ArNOXw%2FQipWU87q5oZ0VSIe0%3D&reserved=0)
3.	Validate the package for structure before submitting the package for certification to Partner Center - [Run Package Validator tool](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-submission%23step-7-validate-the-package-for-structure&data=05%7C02%7Cshefaali.patankar%40microsoft.com%7C6f0c4adf688f42a7107908dcb6c72a8c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638586213737574171%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=9QGRIRe%2FZ%2B1sgUF5JQjY2DeuyxDt8f8lDwpGFEaaBmg%3D&reserved=0) 
4.	Initiate a certification request in Partner Center- [Verified publisher certification process](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fsubmit-for-certification&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390471799%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=zbm2px5VZ2ABLBFcORH6tu1fsC9KWdXkP1XJFyluMzw%3D&reserved=0)
5.	Ensure your connector & plugin files comply to the Marketplace policies here- [5000 Power Platform Connector Policies for Marketplace](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Flegal%2Fmarketplace%2Fcertification-policies%235000-power-platform-connectors&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390477736%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=h0pI5J5buo82xxvnBVmrAN8K%2F%2BK8xZlUdcSwzWoqWq0%3D&reserved=0) and also [1000 Marketplace policies](https://learn.microsoft.com/en-us/legal/marketplace/certification-policies#100-general)(as applicable) 
6.	Test your connector post certification to provide go-live for deployment- [Testing Guidelines](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-testing&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390483743%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=Z8pSnhCArFRzEr373o29qM5h5VbV1nlN2gVbv%2Blk1aE%3D&reserved=0)
7.	If you wish to Update your connector or plugin in future- [Updating Guidelines](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-updates&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390489681%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=CvxTWJFE3bTFeJ3nDJoCSpzvimhVV9go4cBG%2FQ4gMvo%3D&reserved=0)

For any queries or concerns please get in touch with us on [certificationteam@microsoft.com](mailto:certificationteam@microsoft.com)

Many thanks,
Power Platform 
Copilot connector & plugin certification team
